### PR TITLE
APM > .NET > add new setting DD_TRACE_SERVICE_MAPPING

### PR DESCRIPTION
**Pull request by andrewlock**
### What does this PR do?
Documents a new setting in the .NET Tracer : DD_TRACE_SERVICE_MAPPING

### Motivation
We want to document the latest features of the .NET Tracer.

### Preview
https://docs-staging.datadoghq.com/andrewlock/apm-dotnet/service-mapping/tracing/setup_overview/setup/dotnet-core/?tab=environmentvariables#instrumentation
https://docs-staging.datadoghq.com/andrewlock/apm-dotnet/service-mapping/tracing/setup_overview/setup/dotnet-framework/?tab=environmentvariables#instrumentation

### Additional Notes
N/A
---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/tracing/setup_overview/setup/dotnet-core.md [hunk 1]
```diff
@@ -408,6 +408,7 @@ The following table below lists configuration variables that are available for b
...
 | `DD_TRACE_DEBUG`                                    | Enables or disables debug logging. Valid values are: `true` or `false` (default).                                                                                                                                 |
 | `DD_TRACE_HEADER_TAGS`                              | Accepts a map of case-insensitive header keys to tag names and automatically applies matching header values as tags on root spans. (e.g. : `CASE-insensitive-Header:my-tag-name,User-ID:userId`). Available for version 1.18.3+  |
+| `DD_TRACE_SERVICE_MAPPING`                          | Rename services using configuration. Accepts a map of service name keys to rename, and the name to use instead, in the format `[from-key]:[to-name]` e.g. : `mysql:main-mysql-db, mongodb:offsite-mongodb-service`. `from-key` is specific to the integration type, and should exclude the application name prefix. For example, to rename `my-application-sql-server` to `main-db`, use `sql-server:main-db`. Added in version 1.23.0 |
```
Comment (jaycdave88) [inline content/en/tracing/setup_overview/setup/dotnet-core.md:411]: Will this setting also be available to customers via the NuGet API? If so can we please also add that setting similar to how we do for DD_TRACE_GLOBAL_TAGS? 

Overall the text seems good - a bit wordy but can we trim it down at all?

**Review by jaycdave88 — CHANGES_REQUESTED** [review]

File: content/en/tracing/setup_overview/setup/dotnet-core.md (hunk 1 above)
Comment (andrewlock) [reply content/en/tracing/setup_overview/setup/dotnet-core.md:411]: It's not available in _quite_ the same way as `GlobalTags`,  but it is settable via `SetServiceNameMappings()` so I'll add that. 

The issue with trimming it is clarity - I'm happy to reduce it (it was briefer in [the first iteration](https://github.com/DataDog/documentation/pull/9502/commits/41572048e1ad65af224f1ac998c37f2ed5360678)). I added extra clarification about what the `from-key` needs to be, as this is different to the Java implementation. Also, could remove the examples to make it briefer (but I personally like examples in docs 🙂)

Comment (andrewlock) [reply content/en/tracing/setup_overview/setup/dotnet-core.md:411]: Could make this slightly shorter by removing the first set of examples:

> Rename services using configuration. Accepts a map of services to rename, in the format `[from-key]:[to-name]`. The `from-key` is specific to the integration type, and should exclude the application name prefix. For example, to rename `my-application-sql-server` to `main-db`, use `sql-server:main-db`. Added in version 1.23.0`

Comment (jaycdave88) [reply content/en/tracing/setup_overview/setup/dotnet-core.md:411]: The original looks good to me - we can adjust later if needed.

**Review by lucaspimentel — APPROVED** [review]

File: content/en/tracing/setup_overview/setup/dotnet-core.md (hunk 1 above)
Comment (kayayarai) [inline content/en/tracing/setup_overview/setup/dotnet-core.md:411]:
```suggestion
| `DD_TRACE_SERVICE_MAPPING`                          | Rename services using configuration. Accepts a map of service name keys to rename, and the name to use instead, in the format `[from-key]:[to-name]`. For example: `mysql:main-mysql-db, mongodb:offsite-mongodb-service`. `from-key` is specific to the integration type, and should exclude the application name prefix. For example, to rename `my-application-sql-server` to `main-db`, use `sql-server:main-db`. Added in version 1.23.0 |
```

File: content/en/tracing/setup_overview/setup/dotnet-framework.md [hunk 2]
```diff
@@ -252,6 +252,7 @@ The following table lists the supported configuration variables that are availab
...
 | `DD_TRACE_HEADER_TAGS`<br/><br/> `HeaderTags`       | Accepts a map of case-insensitive header keys to tag names and automatically applies matching header values as tags on root spans. (e.g. : `CASE-insensitive-Header:my-tag-name,User-ID:userId`). Added in version 1.18.3      |
 | `DD_TAGS`<br/><br/>`GlobalTags`                     | If specified, adds all of the specified tags to all generated spans (e.g., `layer:api,team:intake`). Added in version 1.17.0                                  |
+| `DD_TRACE_SERVICE_MAPPING`                          | Rename services using configuration. Accepts a map of service name keys to rename, and the name to use instead, in the format `[from-key]:[to-name]` e.g. : `mysql:main-mysql-db, mongodb:offsite-mongodb-service`. `from-key` is specific to the integration type, and should exclude the application name prefix. For example, to rename `my-application-sql-server` to `main-db`, use `sql-server:main-db`. Added in version 1.23.0  |
```
Comment (kayayarai) [inline content/en/tracing/setup_overview/setup/dotnet-framework.md:255]:
```suggestion
| `DD_TRACE_SERVICE_MAPPING`                          | Rename services using configuration. Accepts a map of service name keys to rename, and the name to use instead, in the format `[from-key]:[to-name]`. For example: `mysql:main-mysql-db, mongodb:offsite-mongodb-service`. `from-key` is specific to the integration type, and should exclude the application name prefix. For example, to rename `my-application-sql-server` to `main-db`, use `sql-server:main-db`. Added in version 1.23.0  |
```

**Review by kayayarai — CHANGES_REQUESTED** [review]

**Review by kayayarai — APPROVED** [review]
